src/blockchain/pop/vbk_block_tree.cpp

```diff
@@ -150,7 +150,8 @@ void VbkBlockTree::removePayloads(index_t* index,
                                   const std::vector<payloads_t>& payloads) {
   assert(index);
   cmp_.removePayloads(*index, payloads);
-  if (index->containingContext.back().empty()) {
+  if (!index->containingContext.empty() &&
+      index->containingContext.back().empty()) {
```

Comment:
Why this additional check? Is it possible that back() is empty? Why we don't allow to pop this empty element?

Reply:
it is possible that deque will not have items, and `context.back()` fails because of that